subxt/src/blocks/extrinsic_types.rs

```diff
@@ -264,17 +266,19 @@ where
             .map_err(scale_decode::Error::from)?;
         }
 
-        let call_start_idx = extrinsic_bytes.len() - cursor.len();
+        let call_start_idx = decoded_extrinsic_bytes.len() - cursor.len();
 
         // Decode the pallet index, then the call variant.
-        let cursor = &mut &extrinsic_bytes[call_start_idx..];
+        let cursor = &mut &decoded_extrinsic_bytes[call_start_idx..];
 
         let pallet_index: u8 = Decode::decode(cursor)?;
         let variant_index: u8 = Decode::decode(cursor)?;
 
+        let decoded_extrinsic_bytes: Arc<[u8]> =
+            Arc::from(decoded_extrinsic_bytes.into_boxed_slice());
```

Comment:
I don't why this boxing is needed here, isn't better to use the `From<&[T]> for Arc<[T]> impl` here?

But I guess the underlying implementation still has to copy to the bytes but should be better anyway is my hunch :P

Reply:
I think doing it the way I did it is quite efficient. It does not involve any heap allocations / memcpy of bytes on the heap, because only the pointers on the stack are given control from Vec -> Box -> Arc and that is it. I think the From<&[T]> will create a copy of the bytes on the heap. What do you think?